crmsh/utils.py

```diff
@@ -1221,31 +1221,22 @@
     return ext_cmd_nosudo("%s %s" % (config.core.editor, fname))
 
 
-def edit_file_ext(fname, template=''):
+@contextmanager
+def edit_file_ext(fname):
     '''
     Edit a file via a temporary file.
     Raises IOError on any error.
     '''
-    if not os.path.isfile(fname):
-        s = template
-    else:
-        s = open(fname).read()
-    filehash = hash(s)
-    tmpfile = str2tmp(s)
-    try:
-        try:
-            if edit_file(tmpfile) != 0:
-                return
-            s = open(tmpfile, 'r').read()
-            if hash(s) == filehash:  # file unchanged
-                return
-            f2 = open(fname, 'w')
-            f2.write(s)
-            f2.close()
-        finally:
-            os.unlink(tmpfile)
-    except OSError as e:
-        raise IOError(e)
+    with create_tempfile() as tmpfile:
+        source_data = read_from_file(fname)
+        str2file(source_data, tmpfile)
```

Comment:
`str2file` uses `open_atomic`. It does not make sense to it for writing a tempfile. Something like `shutil.copyfile` is enough.